src/rule_tester.rs

```diff
@@ -31,55 +37,55 @@ impl RuleTester {
     }
 
     fn run_valid_test(&self, valid_test: &RuleTestValid) {
-        let tmp_dir = TempDir::new("valid_test").unwrap();
-        let test_filename = "tmp.rs";
-        fs::write(tmp_dir.path().join(test_filename), &valid_test.code).unwrap();
-        Command::cargo_bin("tree-sitter-lint")
-            .unwrap()
-            .args(["--rule", &self.rule.name])
-            .current_dir(tmp_dir.path())
-            .assert()
-            .success()
-            .stdout(predicate::str::is_empty());
+        let violations = crate::run_for_slice(
```

Comment:
I didn't really wrap my head around when it would make sense to be running in fixing mode or not when doing rule tests

I guess it's an assumed invariant that there are no modified file contents if no violations are reported (when using `--report-fixed-violations`)? So therefore since "valid" test cases assert that there were no violations reported there's no need to eg also run them in fix mode to see if they try and do any unexpected fixing?

And for invalid test cases, it seems like there's no harm in always running them in fixing mode (even if the rule doesn't support fixing) and then verifying that either the actual fixed output matches your specified `output` or that you didn't specify `output` (if you should be required to specify `output` for any test cases that do any fixing? Not sure if ESLint behaves that way or not?)?